File: src/main/java/io/tolgee/constants/ApiScope.java
package io.tolgee.constants;

import io.tolgee.exceptions.NotFoundException;

public enum ApiScope {
    TRANSLATIONS_VIEW("translations.view"),
    TRANSLATIONS_EDIT("translations.edit"),
    KEYS_EDIT("keys.edit"),
    SCREENSHOTS_UPLOAD("screenshots.upload"),
    SCREENSHOTS_DELETE("screenshots.delete"),
    SCREENSHOTS_VIEW("screenshots.view");

    private String value;

    private ApiScope(String value) {
        this.value = value;
    }

    public static ApiScope fromValue(String value) {
        for (ApiScope scope : ApiScope.values()) {
            if (scope.value.equals(value)) {
                return scope;
            }
        }
        throw new NotFoundException(Message.SCOPE_NOT_FOUND);
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }
}

File: backend/data/src/main/kotlin/io/tolgee/model/enums/ApiScope.kt
package io.tolgee.model.enums

import io.tolgee.constants.Message
import io.tolgee.exceptions.NotFoundException

enum class ApiScope(var value: String) {
  TRANSLATIONS_VIEW("translations.view"),
  TRANSLATIONS_EDIT("translations.edit"),
  KEYS_EDIT("keys.edit"),
  SCREENSHOTS_UPLOAD("screenshots.upload"),
  SCREENSHOTS_DELETE("screenshots.delete"),
  SCREENSHOTS_VIEW("screenshots.view"),
  ACTIVITY_VIEW("activity.view");

  companion object {
    fun fromValue(value: String): ApiScope {
      for (scope in values()) {
        if (scope.value == value) {
          return scope
        }
      }
      throw NotFoundException(Message.SCOPE_NOT_FOUND)
    }
  }
}
